autocomplete.py

```diff
@@ -0,0 +1,36 @@
+"""Auto-complete class relying on a modified trie data structure.
+
+original trie: https://github.com/welliam/data-structures/tree/trie-traversal
+"""
+from trie import Trie
+
+
+class AutoCompleter():
+    """Callable autocomplete class takes a list to create word_list."""
+
+    def __init__(self, word_list, max_completions=5):
+        """Initialize the class with a dictionary."""
+        try:
+            self.max_completions = int(max_completions)
+        except ValueError:
+            raise TypeError('max_completions must be an int type.')
+
+        if isinstance(word_list, list):
+            self.word_list = word_list
+        else:
+            try:
+                with open(word_list, 'r') as f:
+                    self.word_list = list(set(f.read().split()))
```

Comment:
There shouldn't be a need to cast this into a set and then a list here -- `f.readlines()` should do the trick. Since you're just iterating over word_list anyway, there's no need to take up the extra memory to use a list Remember, this file could be millions of lines long. A generator is the way to go here.